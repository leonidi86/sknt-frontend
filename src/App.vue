<template>
<div>
  <h1 class="text-xl mb-2">Начало выходного дня (по-порядку)</h1>
  <ul class="flex flex-col items-start pl-2">
    <li
      v-for="(e, i) in preparedList"
      :key="i"
      class="flex items-center gap-1"
      :style="{ color: e.color }"
    >
      <button
        class="text-red-500"
        title="Remove"
        :data-index="i"
        @click.prevent="removeItem(i)"
      >
        <i-ic-baseline-delete-outline />
      </button>
      #{{ i + 1 }} {{ e.title }}
    </li>
  </ul>
</div>
</template>

<script setup lang="ts">
import { computed, ref } from 'vue';

interface Entity {
title: string;
color?: string;
index: number;
enabled: boolean;
}

// Этот список изменять запрещено
const list = ref<Entity[]>([
{
  title: 'позавтракать',
  color: 'red',
  index: 4,
  enabled: true,
},
{
  title: 'пойти гулять',
  color: 'green',
  index: 6,
  enabled: true,
},
{
  title: 'сходить в туалет',
  color: 'brown',
  index: 2,
  enabled: true,
},
{
  title: 'уйти на работу',
  color: 'cyan',
  index: 5,
  enabled: false,
},
{
  title: 'проснуться',
  color: 'blue',
  index: 1,
  enabled: true,
},
{
  title: 'умыться',
  color: 'purple',
  index: 3,
  enabled: true,
},
]);

const preparedList = computed(() => {
return [...list.value]
  .filter((item) => item.enabled)
  .sort((a, b) => a.index - b.index);
});

function removeItem(index: number) {
const itemIndex = list.value.findIndex((item) => item.index === preparedList.value[index].index);
if (itemIndex !== -1) {
  list.value.splice(itemIndex, 1);
}
}
</script>
