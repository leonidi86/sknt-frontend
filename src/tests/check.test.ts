import { expect, test } from 'vitest';
import { mount } from '@vue/test-utils';
import App from '../App.vue';

test('порядок пунктов и их содержимое верное', async () => {
  const wrapper = mount(App, {});

  expect(wrapper.html()).toMatchSnapshot();
});

test('удаление конкретного пункта', async () => {
  const wrapper = mount(App, {});

  expect(wrapper.text()).toContain('проснуться');

  await wrapper.find(`button[data-index="0"]`).trigger('click');

  expect(wrapper.text()).not.toContain('проснуться');
  expect(wrapper.text()).toContain('пойти гулять');
});

test('кол-во пунктов после удаления уменьшилось', async () => {
  /*
   * Задача:
   * 1) написать тест, который проверит, что после клика на значок удаления кол-во элементов уменьшится
   *
   * ПРИМЕЧАНИЕ - тесты запускаются командой npm test
   */

  const wrapper = mount(App, {});
  const initialItemCount = wrapper.findAll('li').length;

  expect(initialItemCount).toBeGreaterThan(0);

  const firstItemIndex = 0;

  await wrapper.find(`button[data-index="${firstItemIndex}"]`).trigger('click');

  const updatedItemCount = wrapper.findAll('li').length;

  expect(updatedItemCount).toBe(initialItemCount - 1);
});
