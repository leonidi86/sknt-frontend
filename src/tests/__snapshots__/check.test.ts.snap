// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`порядок пунктов и их содержимое верное 1`] = `
"<div>
  <h1 class=\\"text-xl mb-2\\">Начало выходного дня (по-порядку)</h1>
  <ul class=\\"flex flex-col items-start pl-2\\">
    <li class=\\"flex items-center gap-1\\" style=\\"color: blue;\\"><button class=\\"text-red-500\\" title=\\"Remove\\" data-index=\\"0\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <path fill=\\"currentColor\\" d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z\\"></path>
        </svg></button> #1 проснуться</li>
    <li class=\\"flex items-center gap-1\\" style=\\"color: brown;\\"><button class=\\"text-red-500\\" title=\\"Remove\\" data-index=\\"1\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <path fill=\\"currentColor\\" d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z\\"></path>
        </svg></button> #2 сходить в туалет</li>
    <li class=\\"flex items-center gap-1\\" style=\\"color: purple;\\"><button class=\\"text-red-500\\" title=\\"Remove\\" data-index=\\"2\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <path fill=\\"currentColor\\" d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z\\"></path>
        </svg></button> #3 умыться</li>
    <li class=\\"flex items-center gap-1\\" style=\\"color: red;\\"><button class=\\"text-red-500\\" title=\\"Remove\\" data-index=\\"3\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <path fill=\\"currentColor\\" d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z\\"></path>
        </svg></button> #4 позавтракать</li>
    <li class=\\"flex items-center gap-1\\" style=\\"color: green;\\"><button class=\\"text-red-500\\" title=\\"Remove\\" data-index=\\"4\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <path fill=\\"currentColor\\" d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z\\"></path>
        </svg></button> #5 пойти гулять</li>
  </ul>
</div>"
`;
